sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCount.java

```diff
@@ -237,10 +240,11 @@ private Builder(HllCountInitFn<InputT, ?> initFn) {
        * PCollection<InputT>} and returns a {@code PCollection<byte[]>} which consists of the HLL++
        * sketch computed from the elements in the input {@code PCollection}.
        *
-       * <p>Returns an empty output {@code PCollection} if the input {@code PCollection} is empty.
+       * <p>Returns a singleton {@code PCollection} with an "empty sketch" (0-length byte array) if
+       * the input {@code PCollection} is empty.
```

Comment:
> Or does a perKey aggregation work exactly like a group-by, i.e., it can never be associated with an empty aggregation?

Yes. (Under the hood it uses a transform called `GroupByKey`.)